### DISCUSSION TITLE
Is it possible to filter the deleted records that are synchronized with each database?

### DISCUSSION THREAD

**Discussion by sjimenezbenito** [Mimetis/Dotmim.Sync#1289] [Q&A] [answered]
When I delete multiple records in the central database, they are replicated to all the 'child' databases ignoring the 'where' condition, even if they do not apply to those databases.

Is it possible to filter the deleted records that are synchronized with each database?

![Delete](https://github.com/user-attachments/assets/9139ff01-5cf5-4676-bdfa-55860471de26)

**Comment by Mimetis** [comment, marked as answer]
The problem with deleted rows, that is not handled by **DMS** correctly, is that when a row is deleted, the only things we have in the tracking tables it the primary key of this row (row to be deleted)
We can't know then if a client needs to delete this row or not, because we can't apply the filter only on a primary key.
That's why every deleted rows are always sent to all clients, just in case.

**Reply by sjimenezbenito** [reply]
That makes perfect sense, thank you for the clarification. I will look for a way to perform these deletions in a "logical" manner and have a scheduled task (or something similar) in the databases, that handles the deletions and cleans up the change table so they don't get replicated.
